sdk/src/azure/core/az_http_policy_retry.c

```diff
@@ -84,67 +75,77 @@ AZ_INLINE AZ_NODISCARD int32_t _az_uint32_span_to_int32(az_span span)
   return value < INT32_MAX ? (int32_t)value : INT32_MAX;
 }
 
+AZ_INLINE AZ_NODISCARD bool _az_http_policy_retry_should_retry_http_response_code(
```

Comment:
Probably outside the scope of this PR, but looks like Storage, in .NET, modifies the retry check to be different than the base implementation:
https://github.com/Azure/azure-sdk-for-net/blob/3065b59a09795d9bf14064c921e80784d34bd63e/sdk/storage/Azure.Storage.Common/src/Shared/StorageResponseClassifier.cs#L23-L44

Also, credential adds 404 as another status  code that triggers a retry:
https://github.com/Azure/azure-sdk-for-net/blob/c8a4acbe0de9eb70f9cb95af35eed5c93484ea88/sdk/identity/Azure.Identity/src/CredentialPipeline.cs#L63-L66

cc @JeffreyRichter

Reply:
Also, the base impl of the Java retry policy is different for some reason:
https://github.com/Azure/azure-sdk-for-java/blob/126f19539f0d232ff1d6ce57bb1b34329f069324/sdk/core/azure-core/src/main/java/com/azure/core/http/policy/RetryStrategy.java#L39-L46

Reply:
Yep - too many opinions on this even in different SDK. Java also does `>= HTTP_INTERNAL_ERROR`, that we can now also do. Go is also different, BTW.
We don't have a credential now, but should we re-add it again - we would need to add `struct { _az_should_retry_fn } _internal` to the az_retry_options; or the auth policy will need to have its own retry policy, which preferably reuses the existing one in some effective manner.